vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/permission_rules/minimum_wait_time_frequency_cap.cc

```diff
@@ -0,0 +1,72 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/frequency_capping/permission_rules/minimum_wait_time_frequency_cap.h"
+#include "bat/ads/internal/frequency_capping/frequency_capping.h"
+#include "bat/ads/ads_client.h"
+#include "bat/ads/internal/time.h"
+#include "bat/ads/internal/client.h"
+#include "bat/ads/internal/ads_impl.h"
+
+#include "bat/ads/ad_info.h"
+
+namespace ads {
+
+bool MinimumWaitTimeFrequencyCap::IsAllowed() {
+  if (ads_->IsMobile()) {
+    return true;
+  }
+
+  auto history = frequency_capping_->GetAdsHistory();
+
+  auto respects_hour_limit = AreAdsPerHourBelowAllowedThreshold(history);
```

Comment:
Why do we have two IsAllowed checks, one for `respects_hour_limit` and `respects_minimum_wait_time` for same frequency cap, can we decouple ads per hour check please

Reply:
Not sure what you mean here... this is the original logic but broken into two functions for single responsibility. How would we decouple this further?

Reply:
As in why have we not got two separate frequency cap classes for these?

Reply:
Discussed and understood - will split into two separate object & tests.

Reply:
Done